features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/TimelineItemEventRow.kt

```diff
@@ -0,0 +1,361 @@
+        Text(
+            text = sender,
+            style = MaterialTheme.typography.titleMedium,
+            modifier = Modifier
+                .alignBy(LastBaseline)
+        )
+    }
+}
+
+@Composable
+private fun MessageEventBubbleContent(
+    event: TimelineItem.Event,
+    interactionSource: MutableInteractionSource,
+    onMessageClick: () -> Unit,
+    onMessageLongClick: () -> Unit,
+    inReplyToClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val isMediaItem = event.content is TimelineItemImageContent || event.content is TimelineItemVideoContent
+    val replyToDetails = event.inReplyTo as? InReplyTo.Ready
+
+    @Composable
+    fun ContentView(
+        modifier: Modifier = Modifier
+    ) {
+        TimelineItemEventContentView(
+            content = event.content,
+            interactionSource = interactionSource,
+            onClick = onMessageClick,
+            onLongClick = onMessageLongClick,
+            modifier = modifier,
+        )
+    }
+
+    @Composable
+    fun ContentAndTimestampView(
+        overlayTimestamp: Boolean,
+        modifier: Modifier = Modifier,
+        contentModifier: Modifier = Modifier,
+        timestampModifier: Modifier = Modifier,
+    ) {
+        if (overlayTimestamp) {
+            Box(modifier) {
+                ContentView(modifier = contentModifier)
+                TimelineEventTimestampView(
+                    event = event,
+                    onClick = onMessageClick,
+                    modifier = timestampModifier
+                        .padding(horizontal = 4.dp, vertical = 4.dp) // Outer padding
+                        .background(LocalColors.current.gray300, RoundedCornerShape(10.0.dp))
+                        .align(Alignment.BottomEnd)
+                        .padding(horizontal = 4.dp, vertical = 2.dp) // Inner padding
+                )
+            }
+        } else {
+            Column(modifier) {
+                ContentView(modifier = contentModifier.padding(start = 12.dp, end = 12.dp, top = 8.dp))
+                TimelineEventTimestampView(
+                    event = event,
+                    onClick = onMessageClick,
+                    modifier = timestampModifier
+                        .align(Alignment.End)
+                        .padding(horizontal = 8.dp, vertical = 2.dp)
+                )
+            }
+        }
+    }
+
+    /** Used only for media items, with no reply to metadata. It displays the contents with no paddings. */
+    @Composable
+    fun SimpleMediaItemLayout(modifier: Modifier = Modifier) {
+        ContentAndTimestampView(overlayTimestamp = true, modifier = modifier)
+    }
+
+    /** Used for every other type of message, groups the different components in a Column with some space between them. */
+    @Composable
+    fun CommonLayout(
+        inReplyToDetails: InReplyTo.Ready?,
+        modifier: Modifier = Modifier
+    ) {
+        EqualWidthColumn(modifier = modifier, spacing = 8.dp) {
```

Comment:
Should we not avoid using `EqualWidthColumn` if the inReply is null?

Reply:
I just changed `EqualWidthColumn` so it doesn't re-measure everything if all the items already have the same width. If we only have 1 child (this case) that should already be the case and it will just work like a simple Column.

Reply:
Ok nice, thanks!